src/components/ScholarshipList.jsx

```diff
@@ -1,43 +1,113 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
-import List from '@material-ui/core/List';
-import ListItem from '@material-ui/core/ListItem';
-import ListItemText from '@material-ui/core/ListItemText';
+import {
+  Button,
+  Card,
+  CardActionArea,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Grid,
+  IconButton,
+  Link as MuiLink,
+  Typography,
+} from '@material-ui/core';
+import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
+import ShareIcon from '@material-ui/icons/Share';
 import { makeStyles } from '@material-ui/core/styles';
 import PropTypes from 'prop-types';
+import { BRAND_NAME } from '../config/constants';
 
 const useStyles = makeStyles((theme) => ({
-  root: {
-    width: '100%',
-    maxWidth: '100%',
-    backgroundColor: theme.palette.background.paper,
+  actions: {
+    padding: theme.spacing(2),
+    paddingTop: theme.spacing(1),
   },
-  link: {
-    textDecoration: 'none',
+  content: {
+    paddingTop: theme.spacing(0),
+    paddingBottom: theme.spacing(1),
   },
-  item: {
-    color: '#000',
+  description: {
+    display: '-webkit-box',
+    '-webkit-line-clamp': 5,
+    lineClamp: 5,
+    '-webkit-box-orient': 'vertical',
+    overflow: 'hidden',
   },
 }));
 
-function ScholarshipList(props) {
+function ScholarshipList({ scholarships }) {
   const classes = useStyles();
-  const { scholarships } = props;
-  const scholarshipList = Object.entries(scholarships).map(([id, item]) => (
-    <Link to={`/scholarships/${id}`} key={id} className={classes.link}>
-      <ListItem divider alignItems="flex-start" className={classes.item}>
-        <ListItemText primary={item.name} secondary={item.school} />
-        <ListItemText primary={item.amount} />
-        <ListItemText primary={item.deadline} />
-      </ListItem>
-    </Link>
-  ));
-
-  return <List className={classes.root}>{scholarshipList}</List>;
+  const shareFn = (id, data) => () => {
+    if (navigator.share) {
```

Comment:
where does `navigator` come from? I don't see it imported?

Reply:
[navigator](https://developer.mozilla.org/en-US/docs/Web/API/Window/navigator) is something present in the Web APIs that represents the state and identity of the user agent.

Reply:
ooo I see!